### ISSUE TITLE
PayPal "refund amount" field is readonly after updating to version 3.6.0

### ISSUE THREAD

**Issue by seanconklin** [woocommerce/woocommerce#23379]
**Describe the bug**
After updating from Woo 3.5.8 to 3.6.0 PayPal orders (built in WooCommerce PayPal not a plugin) are no longer refundable. The field where you enter the amount to refund is greyed out and in inspection is set to readonly.

**To Reproduce**
1. Place an order using PayPal during checkout
2. Inside the WP Admin > WooCommerce > Orders area, edit the order
3. Click the Refund button
4. Notice the refund amount field is greyed out and impossible to use

**Screenshots**

**Expected behavior**
This field should be usable.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated all plugins other than WooCommerce itself.
- [x] This bug happens with Storefront theme.
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```
### WordPress Environment ###

WordPress address (URL): https://jinnybeyer.com
Site address (URL): https://jinnybeyer.com
WC Version: 3.6.1
Log Directory Writable: ✔
WP Version: 5.1.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 7.2.17
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 7.64.1
OpenSSL/1.1.0j

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.1.31-MariaDB
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.6.1
WC Database Prefix: wp_147894347_
MaxMind GeoIP Database: ✔
Total Database Size: 274.64MB
Database Data Size: 179.64MB
Database Index Size: 95.00MB
wp_147894347_woocommerce_sessions: Data: 12.02MB + Index: 2.02MB
wp_147894347_woocommerce_api_keys: Data: 0.00MB + Index: 0.00MB
wp_147894347_woocommerce_attribute_taxonomies: Data: 0.00MB + Index: 0.01MB
wp_147894347_woocommerce_downloadable_product_permissions: Data: 0.00MB + Index: 0.01MB
wp_147894347_woocommerce_order_items: Data: 1.73MB + Index: 0.97MB
wp_147894347_woocommerce_order_itemmeta: Data: 13.27MB + Index: 9.19MB
wp_147894347_woocommerce_tax_rates: Data: 0.00MB + Index: 0.01MB
wp_147894347_woocommerce_tax_rate_locations: Data: 0.00MB + Index: 0.00MB
wp_147894347_woocommerce_shipping_zones: Data: 0.00MB + Index: 0.00MB
wp_147894347_woocommerce_shipping_zone_locations: Data: 0.00MB + Index: 0.01MB
wp_147894347_woocommerce_shipping_zone_methods: Data: 0.00MB + Index: 0.00MB
wp_147894347_woocommerce_payment_tokens: Data: 0.00MB + Index: 0.00MB
wp_147894347_woocommerce_payment_tokenmeta: Data: 0.00MB + Index: 0.00MB
wp_147894347_woocommerce_log: Data: 0.00MB + Index: 0.00MB
wp_147894347_commentmeta: Data: 0.92MB + Index: 0.44MB
wp_147894347_comments: Data: 10.48MB + Index: 4.13MB
wp_147894347_et_divi_ab_testing_stats: Data: 0.02MB + Index: 0.00MB
wp_147894347_links: Data: 0.00MB + Index: 0.00MB
wp_147894347_ms_snippets: Data: 0.02MB + Index: 0.00MB
wp_147894347_options: Data: 3.16MB + Index: 1.03MB
wp_147894347_postmeta: Data: 64.60MB + Index: 30.30MB
wp_147894347_posts: Data: 18.57MB + Index: 1.88MB
wp_147894347_redirection_404: Data: 0.02MB + Index: 0.06MB
wp_147894347_redirection_groups: Data: 0.02MB + Index: 0.03MB
wp_147894347_redirection_items: Data: 0.02MB + Index: 0.09MB
wp_147894347_redirection_logs: Data: 0.02MB + Index: 0.08MB
wp_147894347_snippets: Data: 0.06MB + Index: 0.00MB
wp_147894347_termmeta: Data: 0.07MB + Index: 0.05MB
wp_147894347_terms: Data: 0.04MB + Index: 0.09MB
wp_147894347_term_relationships: Data: 0.32MB + Index: 0.69MB
wp_147894347_term_taxonomy: Data: 0.06MB + Index: 0.08MB
wp_147894347_usermeta: Data: 49.90MB + Index: 39.26MB
wp_147894347_users: Data: 4.00MB + Index: 3.94MB
wp_147894347_wc_download_log: Data: 0.02MB + Index: 0.03MB
wp_147894347_wc_product_meta_lookup: Data: 0.28MB + Index: 0.58MB
wp_147894347_wc_webhooks: Data: 0.02MB + Index: 0.02MB

### Post Type Counts ###

acf: 7
acf-field: 20
acf-field-group: 7
attachment: 4529
block_fabric: 1
block-fabric: 229
custom_css: 2
customize_changeset: 1
display_type: 7
espresso_attendees: 27
espresso_events: 24
espresso_venues: 1
et_pb_layout: 7
jetpack_migration: 2
jp_img_sitemap: 6
jp_img_sitemap_index: 3
jp_sitemap: 3
jp_sitemap_master: 3
nav_menu_item: 24
ngg_gallery: 5
ngg_pictures: 38
page: 54
post: 254
product: 1437
product_variation: 1387
revision: 97
scheduled-action: 15
shop_coupon: 22
shop_order: 8887
shop_order_refund: 1582
was: 4
wdf_scan: 1
wdf_scan_item: 7
woocarousel: 1
woocarouselpro: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (26) ###

Advanced Custom Fields: Repeater Field: by Elliot Condon – 2.1.0
Advanced Custom Fields: by Elliot Condon – 5.7.12
Akismet Anti-Spam: by Automattic – 4.1.1
AMP: by AMP Project Contributors – 1.1.0
Classic Editor: by WordPress Contributors – 1.4
Code Snippets: by Shea Bunge – 2.13.3
Divi Builder: by Elegant Themes – 2.22.3
Enhanced Media Library: by wpUXsolutions – 2.7.2
Flexible Shipping UPS: by WP Desk – 1.4.2 – Not tested with the active version of WooCommerce
Jetpack by WordPress.com: by Automattic – 7.2.1
Redirection: by John Godley – 4.2.3
Run SQL Query: by Fabricio Carella – 1.0.0
Compress JPEG & PNG images: by TinyPNG – 3.2.0
User Switching: by John Blackbourn & contributors – 1.5.0
WooCommerce Advanced Quantity: by Morning Train Technologies ApS – 2.4.3 – Not tested with the active version of WooCommerce
Advanced Order Export For WooCommerce: by AlgolPlus – 2.1.1 – Not tested with the active version of WooCommerce
WooCommerce Authorize.Net Gateway: by SkyVerge – 3.0.2 – Not tested with the active version of WooCommerce
WooCommerce Menu Cart: by Jeremiah Prummer
Ewout Fernhout – 2.7.4 – Not tested with the active version of WooCommerce

WooCommerce Print Invoices/Packing Lists: by SkyVerge – 3.6.3 – Not tested with the active version of WooCommerce
WooCommerce Sequential Order Numbers Pro: by SkyVerge – 1.13.1 – Not tested with the active version of WooCommerce
WooCommerce USPS Shipping: by WooCommerce – 4.4.22 – Not tested with the active version of WooCommerce
WooCommerce - ShipStation Integration: by WooCommerce – 4.1.28 – Not tested with the active version of WooCommerce
WooCommerce Wholesale Prices: by Rymera Web Co – 1.6.5 – Not tested with the active version of WooCommerce
WooCommerce: by Automattic – 3.6.1
Page Duplicator: by Samuel Kowlaski – 0.1.1
Custom Product Tabs for WooCommerce: by YIKES
Inc. – 1.6.9 – Not tested with the active version of WooCommerce


### Inactive Plugins (2) ###

All-in-One WP Migration: by ServMask – 6.91
WP-Optimize: by David Anderson
Ruhani Rabin
Team Updraft – 2.3.3


### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Must Use Plugins (1) ###

WP.com Site Helper: by  –

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔

### WC Pages ###

Shop base: #20144 - /shop/
Cart: #14 - /cart/
Checkout: #20145 - /checkout/
My account: #20146 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Jinny Beyer
Version: 1.1
Author URL: https://codedcommerce.com
Child Theme: ✔
Parent Theme Name: Storefront
Parent Theme Version: 2.4.5
Parent Theme Author URL: https://woocommerce.com/
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Authorize.Net Credit Card ###

Environment: Production
Tokenization Enabled: 
Debug Mode: Off
Webhooks Enabled: ❌
						Not generated

### Action Scheduler ###

Complete: 14
Oldest: 2019-04-17 10:15:00 -0400
Newest: 2019-04-17 10:16:03 -0400

Pending: 1
Oldest: 2019-04-24 10:15:00 -0400
Newest: 2019-04-24 10:15:00 -0400

Canceled: 0
Oldest: –
Newest: –

In-progress: 0
Oldest: –
Newest: –

Failed: 0
Oldest: –
Newest: –
```
</details>

**Comment by seanconklin** [comment]
![Screen Shot 2019-04-18 at 10 23 44 AM](https://user-images.githubusercontent.com/19829621/56380482-175f0480-61c7-11e9-9e72-14a45ccda808.png)

**Comment by mikejolley** [comment]
Hi @seanconklin 

I don't know if it's used differently in PayPal but that field is readonly to prevent reporting breaking due to refunding at order level instead of line item level.

If you refund at line item level (e.g. by refunding X qty or an amount for a line item) that total will be updated automatically. The refund is still possible, and because the refund is done on line items, the reporting and taxes will all be kept correct and in sync. 

tl;dr this is intentional - refund the line items instead.

Thanks!

**Comment by co50** [comment]
This makes ZERO sense at all. If a person orders 6 items, and they need to cancel the entire order we've now got to go line by line and what...zero out every quantity? How exactly do you refund line items and is there a shortcut to do this to the whole order? I hope this isn't for real. 

BTW, this affects Stripe as well, not just PP.

**Comment by tatus1969** [comment]
I just stumbled across this either today, after having used it for years without problems. I often have customers who make multiple orders, and I use to refund them the additional shipping cost through a manual partial refund. This is not a line item, and your decision of having removed the possibility for arbitary refunds is affecting my business. I would have understood if you had made this an opt-in feature with corresponding warning, but just removing it is the wrong approach if you ask me.

**Comment by ultrawebmarketing** [comment]
For those effected by this, you can get around this by adding the following code to your functions.php or, better yet, install the open source [Code Snippets](https://wordpress.org/plugins/code-snippets/) plugin and use that.

```
add_action( 'admin_head', function () { 
	$currentPostType = get_post_type();

	if( $currentPostType != 'shop_order' ) {
		return;
	}
?>
	<script>
		(function($){
			$(document).ready(function(){
				$('#refund_amount').removeAttr('readonly');
			});
		})(jQuery);
	</script>
<?php } );
```

**Comment by ChromeOrange** [comment]
Just to be clear, all refunds now have to be processed manually? 

Maybe someone could update the documentation to reflect this new way of making the store owners life easier?

**Comment by galapogos01** [comment]
How do I vote for this?  Very confusing change with a poor UX.
